source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,296 @@
+      :language: ruby
+      :start-after: start BSON model
+      :end-before: end BSON model
+
+Persistence Context Attributes
+------------------------------
+
+Every model class contains the following methods, which you can use to retrieve
+information about where {+odm+} persists the model:
+
+- ``client_name``: Retrieves the client name
+- ``database_name``: Retrieves the database name
+- ``collection_name``: Retrieves the collection name
+
+The following example shows how to retrieve and print the names of the client, 
+database, and collection where documents for the ``Band`` class are persisted:
+
+.. io-code-block::
+
+  .. input:: /includes/data-modeling/persistence-configuration.rb
+     :language: ruby
+     :start-after: start persistence context attributes
+     :end-before: end persistence context attributes
+
+  .. output::
+     :language: ruby
+     :visible: false
+
+     default
+
+     my_bands
+
+     bands
+
+Customize Your Persistence Configuration
+----------------------------------------
+
+{+odm+} provides both model-level and runtime options for customizing your 
+persistence configuration. The following sections describe these options.
+
+Model-Level Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Perhaps you want to store your model's documents in a collection with a 
+different name than the pluralized form of the model class name.
+
+You can use the ``store_in`` macro to change the collection, database, or client
+where {+odm+} stores a model's documents. The following example shows how
+to use the ``store_in`` macro to store documents from the ``Person`` class in
+a collection called ``citizens`` in the ``other`` database within a client
+named ``analytics``.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in example
+   :end-before: end store_in example
+
+The ``store_in`` macro can also accept a lambda. For example, you might want to 
+use this pattern in a multi-tenant application,
+where multiple users share common access to an application. You can use a lambda
+to perform operations on different databases based on the current thread so that
+users cannot access each others' data. The following example stores documents in
+a database determined by the current thread:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+   :language: ruby
+   :start-after: start store_in lambda example
+   :end-before: end store_in lambda example
+
+.. _mongoid-set-runtime-persistence-options:
+
+Runtime Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``with()`` method on a model class or instance to change
+a model's persistence configuration for a group of operations during runtime.
+
+The persistence context for the model class or instance that you call ``with()``
+on changes to the defined temporary persistence context within a block. 
```

Comment:
S: this sentence gives (good!) information about the `with()` method, but the reader might not have context for it. let's workshop this one together on zoom.